hydra-node/src/Hydra/Ledger/Cardano.hs

```diff
@@ -225,6 +244,148 @@ mkRangedTx (txin, TxOut owner valueIn datum refScript) (recipient, valueOut) sk
+    redeemer :: HashableScriptData <- arbitrary
+    pure $
+      UTxO.fromPairs $
+        concatMap
+          ( \(txin, txOut@(TxOut address value _ refScript)) ->
+              case address of
+                (API.AddressInEra (API.ShelleyAddressInEra _) (ShelleyAddress _ (Ledger.ScriptHashObj _) _)) ->
+                  let script = fromPlutusScript $ Plutus.alwaysSucceedingNAryFunction 3
+                      scriptInfo = ScriptInfo{redeemer, datum, plutusV2Script = script}
+                      scriptHash = hashScript (API.PlutusScript PlutusScriptV2 script)
+                      newAddress =
+                        ShelleyAddressInEra $
+                          makeShelleyAddress
+                            (Testnet $ NetworkMagic 42)
+                            (PaymentCredentialByScript scriptHash)
+                            NoStakeAddress
+                   in [
+                        ( txin
+                        , TxOutWithWitness
+                            { txOut =
+                                TxOut newAddress value (maybe TxOutDatumNone (TxOutDatumHash . hashScriptDataBytes) datum) refScript
+                            , witness = Just scriptInfo
+                            }
+                        )
+                      ]
+                (API.AddressInEra (API.ShelleyAddressInEra _) (ShelleyAddress _ (Ledger.KeyHashObj _) _)) ->
+                  [(txin, TxOutWithWitness{txOut, witness = Nothing})]
+                _ -> []
+          )
+          (UTxO.pairs utxo)
+
+utxoFromTxOutWithWitness :: UTxO' TxOutWithWitness -> UTxO
+utxoFromTxOutWithWitness utxoToCommit =
+  UTxO.fromPairs $
+    (\(txin, TxOutWithWitness{txOut}) -> (txin, txOut))
+      <$> Map.assocs (UTxO.toMap utxoToCommit)
+
+utxoToTxOutWithWitness :: UTxO -> UTxO' TxOutWithWitness
+utxoToTxOutWithWitness utxoToCommit =
+  UTxO.fromPairs $
+    (\(txin, txOut) -> (txin, TxOutWithWitness{txOut, witness = Nothing}))
+      <$> Map.assocs (UTxO.toMap utxoToCommit)
+
+utxoFromTxOutAndWitness :: UTxO' (TxOut CtxUTxO, Witness WitCtxTxIn) -> UTxO
+utxoFromTxOutAndWitness utxoToCommit =
+  UTxO.fromPairs $
+    second fst
+      <$> Map.assocs (UTxO.toMap utxoToCommit)
+
+toUTxOWithWitness :: UTxO -> UTxO' (TxOut CtxUTxO, Witness WitCtxTxIn)
+toUTxOWithWitness utxo =
+  utxo <&> (,KeyWitness KeyWitnessForSpending)
+
+-- | Create _blueprint_ transaction used as a starting point for a proper commit tx.
+--
+-- This is to ensure the backwards compatibility with `UTxO' TxOutWithWitness` which
+-- **was** the request format for the commit endpoint.
+--
+-- By accepting a _blueprint_ tx the user can specify a starting point with any tx fields
+-- they need which gets enriched with commit tx data later and sent back to the user for
+-- signing and _publishing_.
+-- TODO: can this function handle regular UTxO? (e.g. UTxO' TxOut)
+mkBlueprintTx :: UTxO' TxOutWithWitness -> Tx
+mkBlueprintTx utxoWithWitnesses =
```

Comment:
Could: DRY call sites of `mkBlueprintTx` from a normal `UTxO`

I saw the combination of `utxoToTxOutWithWitness` and `mkBlueprintTx` at least 3-4 times in this PR and this suggests that there should be a `mkBlueprintTxFromUTxO :: UTxO -> Tx`.

In general though, we have a the problem of an imprecise API (and lots of inconsistent haddocks) because of the `UTxO` "to commit" vs. `UTxO` "to lookup inputs from" in relation to the blueprint `Tx`. Hence I would suggest a bit more a type driven approach (but still light-weight):

```haskell

draftCommitTx :: Chain tx -> HeadId -> CommitTxDraft tx -> m tx

-- Defined in Hydra.Chain or Hydra.Chain.CommitTxDraft

-- | A nice explanation what a transaction draft is, mentioning that the actual UTxO to commit are the inputs spent by `draftTx`, which are expected to be contained in `lookupUTxO`.
data CommitTxDraft tx = CommitTxDraft { lookupUTxO :: UTxOType tx, draftTx :: tx }

-- Defined in Hydra.API.HTTPServer or Hydra.Chain.Direct...

-- | Simple variant which creates a draft committing all outputs in given UTxO.
mkCommitTxDraft :: UTxO -> CommitTxDraft Tx

-- | Advanced variant which translates from UTxOWithWitnesses to a commit tx draft.
mkCommitTxDraft' :: UTxO' TxOutWithWitness -> CommitTxDraft Tx

-- Example signatures in Hydra.Chain.Direct...

-- NOTE: this does not require a haddock anymore as `ComitTxDraft` is distinguishable from `UTxO`
commit' ::
  ChainContext ->
  HeadId ->
  -- | Spendable 'UTxO'
  UTxO ->
  CommitTxDraft Tx ->
  Either (PostTxError Tx) Tx

commitTx ::
  NetworkId ->
  -- | Published Hydra scripts to reference.
  ScriptRegistry ->
  HeadId ->
  Party ->
  CommitTxDraft Tx ->
  -- | The initial output (sent to each party) which should contain the PT and is
  -- locked by initial script
  (TxIn, TxOut CtxUTxO, Hash PaymentKey) ->
  Tx
```

As you can see I also tried the name "draft tx" to see whether it's clearer instead of "blueprint tx", but any is fine for me.